internal/controller/stas/workload_controller.go

```diff
@@ -118,6 +118,34 @@ func (r *PodReconciler) reconcilePod() reconcile.Func {
 	}
 }
 
+func (r *PodReconciler) imagePodIndex(ctx context.Context, pod *corev1.Pod) (map[string][]corev1.Pod, error) {
```

Comment:
Instead of a function to create an index, which is probably not reusable, could it be easier to make the function/helper more functional? Maybe `findOwnedPods(owner metav1.OwnerReference, containerName string, stasv1alpha.Image image)`?

Reply:
As we _List_ pods and parse images, we should store/keep the information somehow, and not recompute owners for each function call. We could store the relevant info in a struct, but tried to now instead return a `findOwnedPods`-like function directly - instead of a type with such a function.